compiler/rustc_middle/src/thir.rs

```diff
@@ -466,7 +470,7 @@ pub enum ExprKind<'tcx> {
 
 impl<'tcx> ExprKind<'tcx> {
     pub fn zero_sized_literal(user_ty: Option<Canonical<'tcx, UserType<'tcx>>>) -> Self {
-        ExprKind::NonHirLiteral { lit: ty::ScalarInt::ZST, user_ty }
+        ExprKind::ZstLiteral { user_ty }
     }
 }
```

Comment:
remove that function instead? as you can now just use the corresponding variant

Reply:
I wanted to keep the changes local.